File: frontend/types/message.ts
import { User } from './user';

export type MessageStatus = 'SENT' | 'DELIVERED' | 'READ';

export interface MessageReceipt {
  user_id: string;
  status: MessageStatus;
  delivered_at?: string;
  read_at?: string;
}

export interface Message {
  id: string;
  conversation_id: string;
  sender_id: string;
  sender?: User;
  content: string;
  message_type: 'TEXT' | 'SYSTEM';
  created_at: string;
  edited_at?: string;
  deleted_at?: string;
  temp_id?: string;
  receipts?: MessageReceipt[];
}

File: backend/schemas/message.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel
from backend.schemas.user import UserResponse

class MessageReceiptResponse(BaseModel):
    user_id: str
    status: str
    delivered_at: Optional[datetime] = None
    read_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class MessageResponse(BaseModel):
    id: str
    conversation_id: str
    sender_id: str
    sender: Optional[UserResponse] = None
    content: str
    message_type: str = "TEXT"
    created_at: datetime
    edited_at: Optional[datetime] = None
    deleted_at: Optional[datetime] = None
    receipts: List[MessageReceiptResponse] = []

    class Config:
        from_attributes = True

class SendMessageRequest(BaseModel):
    content: str
    message_type: str = "TEXT"

class PaginatedMessagesResponse(BaseModel):
    messages: List[MessageResponse]
    next_cursor: Optional[str] = None
    has_more: bool = False
